db/controller/topics.controller.js

```diff
@@ -51,4 +56,48 @@ exports.getTopics = async (req, res) => {
     } catch (err) {
       next(err);
     }
-  };
\ No newline at end of file
+  };
+
+
+  exports.postComment = async (req, res, next) => {
+    try {
+      const { article_id } = req.params
+      const { username, body } = req.body
+      insertCommentByArticleId(article_id, username, body)
+          .then((comment) => {
+              res.status(201).send({ comment })
+          })
+          .catch((err) => {
+              next(err)
+          })
+    } catch (err) {
+        if (err.code === "23503") {
+            res.status(404).send({ msg: "User not found" });
+        } else if (err.code === "22P02") {
+            res.status(400).send({ msg: "Invalid article_id format" });
+        } else {
+            next(err);
+        }
```

Comment:
This is not how we should be dealing with errors. Have a re-read of the notes on how this should be done instead:

https://l2c.northcoders.com/courses/sd-notes/back-end#sectionId=error-handling,step=intro